nevergrad/optimization/optimizerlib.py

```diff
@@ -1217,7 +1217,7 @@ def __init__(self, instrumentation: Union[int, Instrumentation], budget: Optiona
     def _optimizers(self) -> List[base.Optimizer]:
         if not self._optimizers_:
             self._optimizers_ = []
-            conv = {"num_workers": self.num_workers, "dimension": self.dimension}
+            conv = {"num_workers": self.num_workers, "dimension": self.dimension, "sqrt": int(np.sqrt(self.budget)) if self.budget else self.num_workers}
             budgets = [conv[b] if isinstance(b, str) else b for b in self._parameters.budgets]
```

Comment:
Reading l 1212 I think self._parameters.budgets and Chaining will always be [10] unless you modify it explicitely outside of the methods in _Chain. Then budgets will always be [10] and this conv thing doesn't change anything. Am I missing something ?

Reply:
I added a bunch of asserts displaying self._optimizers and self._parameters, and I am confident
that the value of self._parameters is overwritten correctly.
(that's the magical @jrapin code for ParametrizedFamily, and I am so happy that it is also not trivial for you!)